packages/internal/src/diagnostics/index.ts

```diff
@@ -0,0 +1,137 @@
+import { Project, ts } from 'ts-morph';
+import { init, getGraphQLDiagnostics } from '@0no-co/graphqlsp/api';
+import path from 'path';
+import fs from 'fs';
+import { resolveTypeScriptRootDir } from '../resolve';
+import type { SchemaOrigin } from '../loaders';
+import { load } from '../loaders';
+
+type Severity = 'error' | 'warn' | 'info';
+const severities: Severity[] = ['error', 'warn', 'info'];
+export interface FormattedDisplayableDiagnostic {
+  severity: Severity;
+  message: string;
+  line: number;
+  col: number;
+  file: string | undefined;
+}
+
+type GraphQLSPConfig = {
+  name: string;
+  schema: SchemaOrigin;
+  tadaOutputLocation: string;
+};
+
+export async function check(
+  config: GraphQLSPConfig,
+  minSeverity: Severity = 'error'
+): Promise<FormattedDisplayableDiagnostic[]> {
+  const projectName = path.resolve(process.cwd(), 'tsconfig.json');
+
+  const project = new Project({
+    tsConfigFilePath: './tsconfig.json',
+  });
+
+  const rootPath = (await resolveTypeScriptRootDir(projectName)) || path.dirname(projectName);
+  init({
+    typescript: ts as any,
+  });
+
+  const languageService = project.getLanguageService();
+  const pluginCreateInfo = {
+    config,
+    languageService: {
+      getReferencesAtPosition: (filename, position) => {
+        return languageService.compilerObject.getReferencesAtPosition(filename, position);
+      },
+      getDefinitionAtPosition: (filename, position) => {
+        return languageService.compilerObject.getDefinitionAtPosition(filename, position);
+      },
+      getProgram: () => {
+        const program = project.getProgram();
+        return {
+          ...program,
+          getTypeChecker: () => project.getTypeChecker(),
+          getSourceFile: (s) => {
+            const source = project.getSourceFile(s);
+            return source && source.compilerNode;
+          },
+        };
+      },
+      // This prevents us from exposing normal diagnostics
+      getSemanticDiagnostics: () => [],
+    } as any,
+    languageServiceHost: {} as any,
+    project: {
+      getProjectName: () => path.resolve(process.cwd(), 'tsconfig.json'),
+      readFile: (p) => {
+        return fs.readFileSync(p);
+      },
```

Comment:
TODO for myself: I'll need to look at this closer. This seems a bit odd, since I assume, this is somehow fed from the `Host` accessing FS. In this case, we may have some unintended blocking FS ops here